Source/spells.cpp

```diff
@@ -146,7 +146,7 @@ int GetManaAmount(Player &player, spell_id sn)
 	if (sn == SPL_HEAL || sn == SPL_HEALOTHER) {
 		ma = (spelldata[SPL_HEAL].sManaCost + 2 * player._pLevel - adj);
 	} else if (spelldata[sn].sManaCost == 255) {
-		ma = ((BYTE)player._pMaxManaBase - adj);
+		ma = ((BYTE)(player._pMaxManaBase >> 6) - adj);

```

Comment:
```suggestion
		ma = (player._pMaxManaBase >> 6) - adj;
```
The cast to byte is unnecessary, it might've truncated and wrapped around to 255/253 if it was ever used so makes little difference in practice.

You're right that this is dead code as well. The only spell effects that have this cost are unlearnable, they're all staff effects (Mana, the Magi, the Jester) or runes (only castable by consuming an item). I can't see a code path where this function is ever called for a non-book spell.